src/SchedulerSubsystem/Scheduler.java

```diff
@@ -1,39 +1,81 @@
 import Networking.Messages.SystemMessage;
 import Networking.Receivers.DMA_Receiver;
 import Networking.Transmitters.DMA_Transmitter;
 import com.sun.jdi.InvalidTypeException;
+import org.junit.jupiter.api.*;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
 
 public class Scheduler implements Runnable {
     private final DMA_Transmitter transmitterToFloor;
     private final DMA_Transmitter transmitterToElevator;
     private final DMA_Receiver receiver;
+    private Set<DestinationEvent> floorRequests;
 
     public Scheduler(DMA_Receiver receiver, DMA_Transmitter transmitterToFloor, DMA_Transmitter transmitterToElevator) {
         this.receiver = receiver;
         this.transmitterToElevator = transmitterToElevator;
         this.transmitterToFloor = transmitterToFloor;
+        floorRequests = new HashSet<DestinationEvent>();
     }
 
     /**
-     * process event received from the floor (up or down request)
-     * and sends it to the elevator.
-     * @param destinationEvent a destination event
+     * Process event received from the floor.
+     *
+     * @param destinationEvent Destination event received from floor.
      */
     private void processDestinationEvent(DestinationEvent destinationEvent) {
-        transmitterToElevator.send(destinationEvent);
-
+        // Store event locally to use in scheduling
+        floorRequests.add(destinationEvent);
     }
 
-    private boolean isElevatorStopping(ElevatorStateEvent event) {
-        // TODO: Implement check
-        return true;
+    /**
+     * Returns true if elevator should stop, false otherwise.
+     *
+     * @param e Elevator state to check.
+     * @return True if elevator should stop, false otherwise.
+     */
+    private boolean isElevatorStopping(ElevatorStateEvent e) {
+        // Elevator should stop if tuple (elevator.currentFloor, elevator.Direction)
+        // belongs to the union of scheduler.destinationEvents and elevator.destinationEvents.
+        //
+        // See state diagram of scheduler for additional context.
+
+        // Create new object for union set to avoid funny business of destination events
+        // being added to the scheduler's floor requests!
+        Set<DestinationEvent> union = new HashSet<DestinationEvent>();
+        union.addAll(floorRequests);
+        union.addAll(e.passengerCountMap().keySet());
+
+        return union.contains(new DestinationEvent(e.currentFloor(), getElevatorDirection(e)));
     }
 
+    /**
+     * Gets the direction an elevator is travelling.
+     *
+     * @param e Elevator state to get direction from.
+     * @return Direction elevator is travelling.
+     */
+    private Direction getElevatorDirection(ElevatorStateEvent e) {
+        // TODO(@braeden): should always return a valid direction or throw error
+        //   I'm worried about returning a null value!
+        // TODO(@braeden): should also assert that every direction is same?
```

Comment:
Honestly, I thought that's how it was already tbh. idk if you want to implement that @ArthurAtangana?